PyriteEditor/src/editor/views/Lights/widget_lights.h

```diff
@@ -0,0 +1,290 @@
+						{
+							LightsCollectionView.selectedId = id;
+							LightsCollectionView.selectedLight = ref;
+						}
+						ImGui::PopID();
+
+				};
+
+				// -- Start the widget
+				ImGui::Begin("Lights");
+				{
+
+					// Add the plus sign
+					//if (ImGui::BeginMenuBar())
+					if (ImGui::BeginPopupContextItem("AddLightPopup"))
+					{
+						if (ImGui::Selectable("Point light"))
+						{
+							LightsCollectionView.sourceCollection->Points.push_back({});
+							LightsCollectionView.UpdateEditorFormat();
+						}
+						if (ImGui::Selectable("Spot light")) 
+						{
+							LightsCollectionView.sourceCollection->Directionals.push_back({}); 
+							LightsCollectionView.UpdateEditorFormat();
+						}
+						if (ImGui::Selectable("Directional light"))
+						{
+							LightsCollectionView.sourceCollection->Directionals.push_back({});
+							LightsCollectionView.UpdateEditorFormat();
+						}
+						ImGui::SetNextItemWidth(-FLT_MIN);
+						ImGui::EndPopup();
+					}
+
+
+					{
+						ImGui::PushStyleColor(ImGuiCol_Button, 0);
+						if (ImGui::ImageButton("AddLight", AddIcon.getRawTexture(), ImVec2{ 32,32 }))
+						{
+							ImGui::OpenPopup("AddLightPopup");
+						}
+						ImGui::SameLine();
+						if (ImGui::ImageButton("RemoveLight", RemoveIcon.getRawTexture(), ImVec2{ 32,32 }))
+						{
+							if (LightsCollectionView.selectedLight)
+							{
+								PYR_LOG(LogWidgets, INFO, "Removing light.");
+								LightsCollectionView.sourceCollection->RemoveLight(LightsCollectionView.selectedLight->sourceLight);
+								LightsCollectionView.UpdateEditorFormat();
+							}
+						}
+						ImGui::PopStyleColor();
+					}
+				}
+				ImGui::Separator();
+				{
+					ImGui::BeginChild("left pane", ImVec2(150, 0), ImGuiChildFlags_Border | ImGuiChildFlags_ResizeX);
+					int id = 0;
+					for (pf_Light& pf_light : LightsCollectionView.EditorFormat)
+					{
+						makeSelectable(pf_light.name.c_str(), id++, &pf_light);
+					}
+					ImGui::EndChild();
+				}
+				ImGui::SameLine();
+
+				// Right
+				
+				if (LightsCollectionView.selectedLight)
+				{
+					ImGui::BeginGroup();
+					ImGui::BeginChild("item view", ImVec2(0, -ImGui::GetFrameHeightWithSpacing())); // Leave room for 1 line below us
+					ImGui::Text(LightsCollectionView.selectedLight->name.c_str());
+					ImGui::Separator();
+					inspectLight(LightsCollectionView.selectedLight);
+					ImGui::EndChild(); 
+					ImGui::EndGroup(); 
+				}
+				
+
+				ImGui::End();
+			}
+		
+			void inspectLight(pf_Light* light) {
+
+				if (!light || !light->sourceLight) return;
+
+				ImGui::Checkbox("IsOn", &light->sourceLight->isOn);
+				ImGui::Separator();
+				ImGui::ColorEdit3("Ambiant", &light->sourceLight->ambiant.x);
+				ImGui::ColorEdit3("Diffuse", &light->sourceLight->diffuse.x);
+				ImGui::Separator();
+				switch (light->type)
+				{
+				case pf_Light::DIR:
+				{
+					pyr::DirectionalLight* sourceLight = static_cast<pyr::DirectionalLight*>(light->sourceLight);
```

Comment:
That looks like a `variant` in disguise, but as-is is fine-ish enough

Reply:
it is indeed